tloncorp/landscape-apps
# fix loading of curios/notes from unjoined channels

Fixes #1976, which includes two separate bugs:

1) Users could see an edit menu in the ref, this removes that.
2) Clicking on a gallery item ref for a gallery you haven't joined would take you to a heap detail page where the curio never loads. This PR auto-joins the user to the channel before attempting to load the curio.

This also fixes two other bugs:
1) A similar bug to the second bug listed above, but for notes (can't find an issue for this one).
2) A text overflow issue with author name/time in the heap detail sidebar.

To test:
Using two test ships, create one group on each ship. Have one ship in both groups initially. The other ship will create a group that the first ship is not a member of. In the first group, add a chat channel. In the second group, add a notebook and gallery channel and some content. Post refs from those notebook/gallery channels in the chat channel on the first group from the second ship and click through on the first ship.

Edit:
Also fixed the package lock, it appears that a dependency was added to master using an older version of npm.
